### PR TITLE
Add editors for Trino database connector

### PR DESCRIPTION
**Pull request by jinanisha**
## Summary
Add UI support for Trino database type, Trino datasource spec, and Trino Delegated Kerberos Authentication Strategy

Corresponding legend-engine changes : https://github.com/finos/legend-engine/pull/1464

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)
Capture

### CODE REVIEW

**Comment by YannanGao-gs** [comment]
Could you please improve the summary and screenshot if possible?

File: packages/legend-graph/src/graphManager/action/changeDetection/STO_Relational_ObserverHelper.ts [hunk 1]
```diff
@@ -37,6 +38,8 @@ import {
   SnowflakeDatasourceSpecification,
   StaticDatasourceSpecification,
   SpannerDatasourceSpecification,
+  TrinoDatasourceSpecification,
+  type TrinoSslSpecification,
```
Comment (YannanGao-gs) [inline packages/legend-graph/src/graphManager/action/changeDetection/STO_Relational_ObserverHelper.ts:42]: just wanna be more consistent, would you mind moving it up?

File: packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/ConnectionEditorState.ts [hunk 2]
```diff
@@ -265,6 +272,14 @@ export class RelationalDatabaseConnectionValueState extends ConnectionValueState
         dataSpec = new SpannerDatasourceSpecification('', '', '', '', '');
         break;
       }
+      case CORE_DATASOURCE_SPEC_TYPE.TRINO: {
+        dataSpec = new TrinoDatasourceSpecification(
+          '',
+          NaN,
```
Comment (YannanGao-gs) [inline packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/ConnectionEditorState.ts:278]: why do we use NaN here?

Comment (jinanisha) [reply packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/ConnectionEditorState.ts:278]: This filed is of number type and we dont want to give any default value, thats why NaN. Is there a better way to achieve the same ?

Comment (YannanGao-gs) [reply packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/ConnectionEditorState.ts:278]: I personally don't prefer NaN. Maybe we could give a dummy default value between 1024 and 65535

**Comment by jinanisha** [comment]
> could you please add one test for this?

File: packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/ConnectionEditorState.ts (hunk 2 above)
Comment (YannanGao-gs) [reply packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/ConnectionEditorState.ts:278]: or maybe undefined?

File: packages/legend-graph/src/graphManager/protocol/pure/v1/model/packageableElements/store/relational/connection/V1_DatasourceSpecification.ts [hunk 3]
```diff
@@ -194,3 +194,42 @@ export class V1_LocalH2DataSourceSpecification
...
+  sslSpecification!: V1_TrinoSslSpecification;
+  catalog?: string | undefined;
+  schema?: string | undefined;
+  clientTags?: string | undefined;
+
+  get hashCode(): string {
+    return hashArray([
+      CORE_HASH_STRUCTURE.TRINO_DATASOURCE_SPECIFICATION,
+      this.host,
+      this.port ?? '',
```
Comment (YannanGao-gs) [inline packages/legend-graph/src/graphManager/protocol/pure/v1/model/packageableElements/store/relational/connection/V1_DatasourceSpecification.ts:228]: port type is number you should use toString()

File: packages/legend-graph/src/graphManager/protocol/pure/v1/model/packageableElements/store/relational/connection/V1_DatasourceSpecification.ts [hunk 4]
```diff
@@ -194,3 +194,42 @@ export class V1_LocalH2DataSourceSpecification
...
+    ]);
+  }
+}
+
+export class V1_TrinoDatasourceSpecification
+  extends V1_DatasourceSpecification
+  implements Hashable
+{
+  host!: string;
+  port!: number | undefined;
```
Comment (YannanGao-gs) [inline packages/legend-graph/src/graphManager/protocol/pure/v1/model/packageableElements/store/relational/connection/V1_DatasourceSpecification.ts:218]: it should be port?

Comment (YannanGao-gs) [inline packages/legend-graph/src/graphManager/protocol/pure/v1/model/packageableElements/store/relational/connection/V1_DatasourceSpecification.ts:218]: you should remove `!` then make it optional

File: packages/legend-graph/src/graph/metamodel/pure/packageableElements/store/relational/connection/DatasourceSpecification.ts [hunk 5]
```diff
@@ -276,3 +276,56 @@ export class SpannerDatasourceSpecification
...
+    this.host = host;
+    this.port = port;
+    this.sslSpecification = sslSpecification;
+  }
+
+  get hashCode(): string {
+    return hashArray([
+      CORE_HASH_STRUCTURE.TRINO_DATASOURCE_SPECIFICATION,
+      this.host,
+      this.port ?? '',
```
Comment (YannanGao-gs) [inline packages/legend-graph/src/graph/metamodel/pure/packageableElements/store/relational/connection/DatasourceSpecification.ts:324]: tostring()

File: packages/legend-graph/src/graph/metamodel/pure/packageableElements/store/relational/connection/DatasourceSpecification.ts [hunk 6]
```diff
@@ -276,3 +276,56 @@ export class SpannerDatasourceSpecification
...
+    ]);
+  }
+}
+
+export class TrinoDatasourceSpecification
+  extends DatasourceSpecification
+  implements Hashable
+{
+  host: string;
+  port: number | undefined;
```
Comment (YannanGao-gs) [inline packages/legend-graph/src/graph/metamodel/pure/packageableElements/store/relational/connection/DatasourceSpecification.ts:303]: use port?

File: packages/legend-graph/src/graphManager/protocol/pure/v1/transformation/pureGraph/to/helpers/V1_RelationalConnectionBuilderHelper.ts [hunk 7]
```diff
@@ -239,6 +244,35 @@ export const V1_buildDatasourceSpecification = (
       protocol.proxyPort,
     );
     return spannerSpec;
+  } else if (protocol instanceof V1_TrinoDatasourceSpecification) {
+    assertNonEmptyString(
+      protocol.host,
+      `Trino datasource specification 'host' field is missing or empty`,
+    );
+    assertNonNullable(
```
Comment (YannanGao-gs) [inline packages/legend-graph/src/graphManager/protocol/pure/v1/transformation/pureGraph/to/helpers/V1_RelationalConnectionBuilderHelper.ts:252]: port is number | undefined now

**Review by YannanGao-gs — APPROVED** [review]

**Comment by MauricioUyaguari** [comment]
@jinanisha  looking at the screenshot, the editor doesnt show an input for kerberos remote service name. 
Why is this ?

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**